nestjs/src/courses/team-distribution/distribute-students.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Team, TeamDistributionStudent } from '@entities/index';
+import { TeamDistribution } from '@entities/teamDistribution';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { shuffle } from 'lodash';
+import { DataSource, Repository } from 'typeorm';
+import { TeamDistributionStudentService } from './team-distribution-student.service';
+import { TeamService } from './team.service';
+
+@Injectable()
+export class DistributeStudentsService {
+  constructor(
+    @InjectRepository(TeamDistribution)
+    private repository: Repository<TeamDistribution>,
+    private teamDistributionStudentService: TeamDistributionStudentService,
+    private teamService: TeamService,
+    private dataSource: DataSource,
+  ) {}
+
+  public getById(id: number) {
+    return this.repository.findOneOrFail({ where: { id } });
+  }
+
+  private getTeamCapacity(teams: Team[], teamSize: number) {
+    return teams.reduce((acc, curr) => acc + teamSize - curr.students.length, 0);
+  }
+
+  private async removeTeams(smallTeams: Team[], teamDistributionId: number, courseId: number) {
```

Comment:
Curious why did you named it `smallTeams` here while in other places it's just `teams` :D

Reply:
fixed